### PR TITLE
feat: mixpanel events for cosmos

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

Add events for
- Withdraw
- Claim
- Deposit

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

## Risk

small risk

### Operations

Check cosmos staking still works for (claim, deposit, withdraw)

### CODE REVIEW

File: src/features/defi/providers/cosmos/components/CosmosManager/Deposit/components/Deposit.tsx [hunk 1]
```diff
@@ -46,10 +51,19 @@ export const Deposit: React.FC<DepositProps> = ({
...
+  const { chainId, assetReference, contractAddress } = query
   const assetNamespace = 'slip44'
+  const assets = useAppSelector(selectAssets)
   const assetId = toAssetId({ chainId, assetNamespace, assetReference })
 
+  const validatorId = toValidatorId({ chainId, account: contractAddress })
+
+  const opportunityMetadataFilter = useMemo(() => ({ validatorId }), [validatorId])
+
+  const opportunityData = useAppSelector(state =>
```
Comment (gomesalexandre) [inline src/features/defi/providers/cosmos/components/CosmosManager/Deposit/components/Deposit.tsx:63]:
```suggestion
  const opportunityMetadata = useAppSelector(state =>
```

Comment (gomesalexandre) [reply src/features/defi/providers/cosmos/components/CosmosManager/Deposit/components/Deposit.tsx:63]: We could also remove the apy now that we're selecting opportunity (meta)data, if you feel like it @reallybeard

File: src/features/defi/providers/cosmos/components/CosmosManager/Deposit/components/Status.tsx [hunk 2]
```diff
@@ -26,19 +35,53 @@ export const Status = () => {
...
 
   const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const assetMarketData = useAppSelector(state => selectMarketDataById(state, assetId))
   if (!asset) throw new Error(`Asset not found for AssetId ${assetId}`)
 
+  const validatorId = toValidatorId({ chainId, account: contractAddress })
+
+  const opportunityMetadataFilter = useMemo(() => ({ validatorId }), [validatorId])
+
+  const opportunityData = useAppSelector(state =>
```
Comment (gomesalexandre) [inline src/features/defi/providers/cosmos/components/CosmosManager/Deposit/components/Status.tsx:51]:
```suggestion
  const opportunityMetadata = useAppSelector(state =>
```

File: src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/components/Confirm.tsx [hunk 3]
```diff
@@ -54,6 +60,13 @@ export const Confirm: React.FC<ConfirmProps> = ({ onNext, accountId }) => {
   const wallet = useWallet().state.wallet
 
   const assetNamespace = 'slip44' // TODO: add to query, why do we hardcode this?
+  const validatorId = toValidatorId({ chainId, account: contractAddress })
+
+  const opportunityMetadataFilter = useMemo(() => ({ validatorId }), [validatorId])
+
+  const opportunityData = useAppSelector(state =>
```
Comment (gomesalexandre) [inline src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/components/Confirm.tsx:67]:
```suggestion
  const opportunityMetadata = useAppSelector(state =>
```

File: src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/components/Status.tsx [hunk 4]
```diff
@@ -28,18 +37,29 @@ export const Status: React.FC<StatusProps> = ({ accountId }) => {
...
+  const { chainId, assetReference, contractAddress } = query
 
   const userAddress: string | undefined = accountId && fromAccountId(accountId).account
 
   const assetNamespace = 'slip44'
+  const validatorId = toValidatorId({ chainId, account: contractAddress })
+
+  const opportunityMetadataFilter = useMemo(() => ({ validatorId }), [validatorId])
+
+  const opportunityData = useAppSelector(state =>
```
Comment (gomesalexandre) [inline src/features/defi/providers/cosmos/components/CosmosManager/Withdraw/components/Status.tsx:49]:
```suggestion
  const opportunityMetadata = useAppSelector(state =>
```

**Review by gomesalexandre — APPROVED** [review]
Tested locally, confirmed this does what it says on the box 🍪

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**